qiskit_optimization/algorithms/qrao/quantum_random_access_optimizer.py

```diff
@@ -0,0 +1,317 @@
+        """Set the rounding scheme."""
+        self._rounding_scheme = rounding_scheme
+
+    def get_compatibility_msg(self, problem: QuadraticProgram) -> str:
+        """Checks whether a given problem can be solved with this optimizer.
+
+        Checks whether the given problem is compatible, i.e., whether the problem can be converted
+        to a QUBO, and otherwise, returns a message explaining the incompatibility.
+
+        Args:
+            problem: The optimization problem to check compatibility.
+
+        Returns:
+            A message describing the incompatibility.
+        """
+        return QuadraticProgramToQubo.get_compatibility_msg(problem)
+
+    def solve_relaxed(
+        self,
+        encoding: QuantumRandomAccessEncoding,
+    ) -> tuple[MinimumEigensolverResult, RoundingContext]:
+        """Solve the relaxed Hamiltonian given by the encoding.
+
+        .. note::
+            This method uses the encoding instance given as ``encoding`` and
+            ignores :meth:`max_vars_per_qubit`.
+
+        Args:
+            encoding: An encoding instance for which :meth:`~QuantumRandomAccessEncoding.encode`
+                has already been called so it has been encoded with a :class:`~.QuadraticProgram`.
+
+        Returns:
+            The result of the minimum eigensolver, and the rounding context.
+
+        Raises:
+            ValueError: If the encoding has not been encoded with a :class:`~.QuadraticProgram`.
+        """
+        if not encoding.frozen:
+            raise ValueError(
+                "The encoding must call ``encode()`` with a ``QuadraticProgram`` before being passed"
+                "to the QuantumRandomAccessOptimizer."
+            )
+
+        # Get the list of operators that correspond to each decision variable.
+        variable_ops = [encoding._term2op(i) for i in range(encoding.num_vars)]
+
+        # Solve the relaxed problem
+        start_time_relaxed = timeit.default_timer()
+        relaxed_result = self.min_eigen_solver.compute_minimum_eigenvalue(
+            encoding.qubit_op, aux_operators=variable_ops
+        )
+        relaxed_result.time_taken = timeit.default_timer() - start_time_relaxed
+
+        # Get auxiliary expectation values for rounding.
+        if relaxed_result.aux_operators_evaluated is not None:
+            expectation_values = [v[0] for v in relaxed_result.aux_operators_evaluated]
+        else:
+            expectation_values = None
+
+        # Get the circuit corresponding to the relaxed solution.
+        if isinstance(relaxed_result, VariationalResult):
+            circuit = relaxed_result.optimal_circuit.bind_parameters(relaxed_result.optimal_point)
+        elif isinstance(self.min_eigen_solver, NumPyMinimumEigensolver):
+            statevector = relaxed_result.eigenstate
+            circuit = QuantumCircuit(encoding.num_qubits)
+            circuit.initialize(statevector)
+        else:
```

Comment:
This is just a comment based on #537 which will need to done for the QRAO logic too in this PR ahead of it being merged. The instance checks here - that they may work only with qiskit_algorithms once things are changed is fine I think - this is new code being introduced and end users will not have code existing with qiskit.algorithms using this to worry about any migration/transition. That qiskit_algorithms must be used for QRAO is ok.